conda_anaconda_telemetry/hooks.py

```diff
@@ -238,11 +241,22 @@ def _conda_request_headers() -> Sequence[HeaderWrapper]:
     return custom_headers
 
 
+@functools.lru_cache(None)
+def should_submit_request_headers(host: str, path: str) -> bool:
+    """Return a list of hosts and paths we want to submit request headers to."""
+    return any(
+        host == host_pattern and (not path_pattern or path.startswith(path_pattern))
+        for host_pattern, path_pattern in REQUEST_HEADER_PATTERNS
+    )
+
+
 @hookimpl
-def conda_session_headers(host: str) -> Iterator[CondaRequestHeader]:
+def conda_request_headers(host: str, path: str) -> Iterator[CondaRequestHeader]:
     """Return a list of custom headers to be included in the request."""
     try:
-        if context.plugins.anaconda_telemetry and host in REQUEST_HEADER_HOSTS:
+        if context.plugins.anaconda_telemetry and should_submit_request_headers(
+            host, path
+        ):
             yield from validate_headers(_conda_request_headers())
```

Comment:
Is `_conda_request_headers()` properly cached? And can we cache the `validate_headers` output instead?

I know, I know, premature optimization, but as long as I'm here...